cmd/kubectl_token.go

```diff
@@ -175,76 +173,107 @@ func deleteCachedCredential(ctx *cli.Context) error {
 	if len(ctx.Args()) == 0 {
 		return cli.ShowSubcommandHelp(ctx)
 	}
-	dir, err := os.Getwd()
+
+	sc, err := lookupServerConfig(ctx)
 	if err != nil {
 		return err
 	}
-	cacheDir := filepath.Join(dir, kubeConfigCache)
+
+	// dir is always set by global default.
+	dir := ctx.GlobalString("config")
+
+	if len(sc.KubeCredentials) == 0 {
+		customPrint(fmt.Sprintf("there are no cached tokens in [%s]", dir))
+		return nil
+	}
+
 	if ctx.Args().First() == "all" {
-		customPrint(fmt.Sprintf("removing cached tokens [%s]", cacheDir))
-		return os.RemoveAll(cacheDir)
+		customPrint(fmt.Sprintf("removing cached tokens [%s]", dir))
+		sc.KubeCredentials = make(map[string]*config.ExecCredential)
+
 	}
 	for _, key := range ctx.Args() {
-		cachePath := filepath.Join(cacheDir, fmt.Sprintf("%s%s", key, cachedFileExt))
-		customPrint(fmt.Sprintf("removing [%s]", cachePath))
-		err := os.Remove(cachePath)
-		if err != nil && !os.IsNotExist(err) {
-			return err
-		}
+		customPrint(fmt.Sprintf("removing [%s]", dir))
+		sc.KubeCredentials[key] = nil
 	}
+
 	return nil
 }
 
-func loadCachedCredential(key string) (*ExecCredential, error) {
-	dir, err := os.Getwd()
+func loadCachedCredential(ctx *cli.Context, key string) (*config.ExecCredential, error) {
+	cred, err := lookupKubeToken(ctx, key)
 	if err != nil {
 		return nil, err
 	}
-	cachePath := filepath.Join(dir, kubeConfigCache, fmt.Sprintf("%s%s", key, cachedFileExt))
-	f, err := os.Open(cachePath)
-	if err != nil {
-		if os.IsNotExist(err) {
-			return nil, nil
-		}
 
-		return nil, err
+	return cred, nil
```

Comment:
we shouldn't return the credential if it's expired, are we checking this elsewhere? Earlier, it was a part of loadCachedCredential https://github.com/rancher/cli/pull/291/files#diff-06991da8e038887c429ccab4b06868704919c15ade73fca3cd9c4fc8997f7138L217-L219

Reply:
I missed this, many apologies.